phoenix-core/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -2586,4 +2588,18 @@ public static boolean isNoUpgradeSet(Properties props) {
     public static void doNotUpgradeOnFirstConnection(Properties props) {
         props.setProperty(DO_NOT_UPGRADE, String.valueOf(true));
     }
+
+    //When upgrading to Phoenix 4.16, make each existing table's DDL timestamp equal to its last
+    // updated row timestamp.
+    public static void bootstrapLastDDLTimestamp(PhoenixConnection metaConnection) throws SQLException  {
+        String pkCols = TENANT_ID + ", " + TABLE_SCHEM +
```

Comment:
Can we add a log before and after this `UPSERT SELECT` to help track the upgrade progress?